### PR TITLE
Fixes #22892 - allow extending audited associations

### PR DESCRIPTION
**Pull request by ares**
I do not see an easy way of adding test. I'd need new full AR model so that I can use audited and then reopen the class to add additional associations.

Here's an example of using this new method - https://github.com/theforeman/foreman_ansible/pull/143

### CODE REVIEW

File: app/models/concerns/audit_associations.rb [hunk 1]
```diff
@@ -50,13 +50,22 @@ module ClassMethods
...
+          configure_dirty_associations(Array(options[:associations]))
         end
 
         super
       end
+
+      def audit_associations(associations)
+        new_associations = Array(associations)
+        configure_dirty_associations(new_associations)
+        self.audited_options[:associations] = Array(self.audited_options[:associations]) + new_associations
```
Comment (tbrisker) [inline app/models/concerns/audit_associations.rb]: I think this should be `|` instead of `+` in case two plugins try to add the same associations.

Comment (ares) [reply app/models/concerns/audit_associations.rb]: good point, added `|`

**Review by tbrisker — APPROVED** [review]
Failures are due to new audited version and mysql issue, since this touches on auditing i would prefer to wait for stable tests before merging.

**Comment by ares** [comment]
[test foreman] [test katello]

**Comment by mmoll** [comment]
@ares please rebase :)

**Comment by ares** [comment]
rebased, should get green now

**Review by tbrisker — APPROVED** [review]
LGTM, Thanks @ares !

**Pull request merged**